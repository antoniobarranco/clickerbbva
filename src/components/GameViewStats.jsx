import React from 'react';
import PropTypes from 'prop-types';

function GameViewStats({
  clickersIncrement, autoClickerCost, megaClickerCost, timeInterval,
}) {
  return (
    <div className="d-grid gap-2 col-8  my-5 mx-auto">
      <ul className="list-group">
        <li className="list-group-item d-flex justify-content-between align-items-center active">
          Game Stats
        </li>
        <li className="list-group-item d-flex justify-content-between align-items-center">
          Increment every
          {' '}
          {timeInterval}
          ms
          <span className="badge bg-primary rounded-pill">{clickersIncrement}</span>
        </li>
        <li className="list-group-item d-flex justify-content-between align-items-center">
          AutoClickers cost
          {' '}
          <span className="badge bg-primary rounded-pill">{autoClickerCost}</span>
        </li>
        <li className="list-group-item d-flex justify-content-between align-items-center">
          MegaClickers cost
          {' '}
          <span className="badge bg-primary rounded-pill">{megaClickerCost}</span>
        </li>
      </ul>
    </div>

  );
}

GameViewStats.propTypes = {
  clickersIncrement: PropTypes.number.isRequired,
  autoClickerCost: PropTypes.number.isRequired,
  megaClickerCost: PropTypes.number.isRequired,
  timeInterval: PropTypes.number.isRequired,
};

export default GameViewStats;
