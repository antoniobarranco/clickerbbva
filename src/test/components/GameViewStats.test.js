import renderer from 'react-test-renderer';
import { render, screen } from "@testing-library/react";
import GameViewStats from "../../components/GameViewStats";

describe('Tests on GameViewStats component', () => {

    test('Test GameViewStats with Snapshot test for UI changes', () => {

        const clickersIncrement = 14;
        const autoClickerCost = 500;
        const megaClickerCost = 2000;
        const timeInterval = 100;

        const tree = renderer.create(
            <GameViewStats
                clickersIncrement={clickersIncrement}
                autoClickerCost={autoClickerCost}
                megaClickerCost={megaClickerCost}
                timeInterval={timeInterval}
            />).toJSON();

        expect(tree).toMatchSnapshot();

    })

    test('Test GameViewStats component shows propertly with their text prop', () => {

        const clickersIncrement = 14;
        const autoClickerCost = 500;
        const megaClickerCost = 2000;
        const timeInterval = 100;

        render(<GameViewStats
                clickersIncrement={clickersIncrement}
                autoClickerCost={autoClickerCost}
                megaClickerCost={megaClickerCost}
                timeInterval={timeInterval}
        />);

        expect(screen.getByTestId('clickersIncrement')).toHaveTextContent(clickersIncrement);
        expect(screen.getByTestId('autoClickerCost')).toHaveTextContent(autoClickerCost);
        expect(screen.getByTestId('megaClickerCost')).toHaveTextContent(megaClickerCost);

    })

})