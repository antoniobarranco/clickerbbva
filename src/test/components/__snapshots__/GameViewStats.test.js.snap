// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GameViewStats component Test GameViewStats with Snapshot test for UI changes 1`] = `
<div
  className="d-grid gap-2 col-8  my-5 mx-auto"
>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item d-flex justify-content-between align-items-center active"
    >
      Game Stats
    </li>
    <li
      className="list-group-item d-flex justify-content-between align-items-center"
    >
      Increment every
       
      100
      ms
      <span
        className="badge bg-primary rounded-pill"
        data-testid="clickersIncrement"
      >
        14
      </span>
    </li>
    <li
      className="list-group-item d-flex justify-content-between align-items-center"
    >
      AutoClickers cost
       
      <span
        className="badge bg-primary rounded-pill"
        data-testid="autoClickerCost"
      >
        500
      </span>
    </li>
    <li
      className="list-group-item d-flex justify-content-between align-items-center"
    >
      MegaClickers cost
       
      <span
        className="badge bg-primary rounded-pill"
        data-testid="megaClickerCost"
      >
        2000
      </span>
    </li>
  </ul>
</div>
`;
